sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.kinesis;
+
+
+import org.apache.beam.sdk.io.kinesis.client.KinesisClientProvider;
+import org.apache.beam.sdk.io.kinesis.client.response.KinesisRecord;
+import org.apache.beam.sdk.io.kinesis.source.KinesisSource;
+import org.apache.beam.sdk.io.kinesis.source.checkpoint.StartingPoint;
+import org.apache.beam.sdk.transforms.PTransform;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.internal.StaticCredentialsProvider;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.kinesis.AmazonKinesis;
+import com.amazonaws.services.kinesis.AmazonKinesisClient;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.InitialPositionInStream;
+import java.util.Date;
+
+/**
+ * {@link PTransform}s for reading from
+ * <a href="https://aws.amazon.com/kinesis/">Kinesis</a> streams.
+ */
+public class KinesisIO {
```

Comment:
Usually, `KinesisIO` class-level Javadoc is where we have the copious Javadoc information that would say how this connector is used. This is the information you put in the `README.md` -- how to construct the readers, etc. It's important to have here since most people will be using IDEs to edit and not looking at github :).

This is also where you would talk about timestamps -- in this case, that records are always given the processing time as their timestamp.

See KafkaIO.java for an example?